app/src/main/java/org/simple/clinic/editpatient/PatientEditScreenController.kt

```diff
@@ -392,7 +364,8 @@ class PatientEditScreenController @Inject constructor(
         .withLatestFrom(ongoingEntryChanges) { _, entry -> entry }
         .withLatestFrom(events.ofType<PatientEditScreenCreated>())
         .map { (ongoingEntry, screenCreated) ->
-          hasEdits(screenCreated.patient, screenCreated.phoneNumber, screenCreated.address, ongoingEntry, OngoingEditPatientEntry.from(screenCreated, dateOfBirthFormatter))
+          val savedEntry = OngoingEditPatientEntry.from(screenCreated, dateOfBirthFormatter)
```

Comment:
Since this will never change for a screen, we can extract it into a separate stream and use a `withLatestFrom` to make the chain smaller.

```kotlin
val savedOngoingEntry = events
        .mapType<PatientEditScreenCreated, OngoingEditPatientEntry> { OngoingEditPatientEntry.from(it, dateOfBirthFormatter) }
    
    val ongoingEntryChanges = events
        .ofType<OngoingEditPatientEntryChanged>()
        .map { it.ongoingEditPatientEntry }

    val hasEntryChangedStream = events
        .ofType<PatientEditBackClicked>()
        .withLatestFrom(ongoingEntryChanges) { _, entry -> entry }
        .withLatestFrom(savedOngoingEntry)
        .map { (ongoingEntry, savedOngoingEntry) -> ongoingEntry != savedOngoingEntry }
```

What do you think?